spring-apps/autosleep-core/src/main/java/org/cloudfoundry/autosleep/ui/web/controller/AutoEnrollmentController.java

```diff
@@ -0,0 +1,143 @@
+
+    @Autowired
+    private EnrollmentValidator validator;
+
+    @Autowired
+    private CloudFoundryApiService cloudfoundryApi;
+
+    @Autowired
+    private ParameterReader<Config.ServiceInstanceParameters.Enrollment> autoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Pattern> excludeFromAutoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Duration> idleDurationReader;
+
+    @Autowired
+    private ParameterReader<EnrollmentConfig.EnrollmentParameters.EnrollmentState> stateReader;
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> getEnrolledOrganization(
+            @PathVariable String organizationId) {
+        log.debug("getEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<OrgEnrollmentConfig>(orgConfig, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> deleteOrganization(
+            @PathVariable String organizationId) {
+        log.debug("deleteEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        orgEnrollmentRepository.delete(organizationId);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> enrolOrganization(
+            @PathVariable String organizationId,
+            @RequestBody OrgEnrollmentConfigRequest orgEnrollerConfigRequest, BindingResult result)
+            throws CloudFoundryException, BindException {
+
+        log.debug("enrolOrganization - {}", organizationId);
+        if (!cloudfoundryApi.isValidOrganization(organizationId)) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        validator.validate(orgEnrollerConfigRequest, result);
+        if (result.hasErrors()) {
+            throw new BindException(result);
+        }
+
+        OrgEnrollmentConfig orgEnrollerConfig = OrgEnrollmentConfig.builder()
+                .organizationGuid(organizationId)
+                .idleDuration(orgEnrollerConfigRequest.getIdleDuration() == null ? null
+                        : idleDurationReader
+                                .readParameter(orgEnrollerConfigRequest.getIdleDuration(), false))
+                .excludeSpacesFromAutoEnrollment(
+                        orgEnrollerConfigRequest.getExcludeSpacesFromAutoEnrollment() == null ? null
+                                : excludeFromAutoEnrollmentReader
```

Comment:
`>> fair amount of boilerplate code in this PR that would be removed/simplified`

The spring validator framework provides for constraints such as notnull, notempty, past, future etc but not validations such as duration validations or pattern compile checks (it provides pattern.match checks though but that does not help us). For these constraints, the framework can return a list of all violations. For our use case, we have to register the validator that execute our custom checks and returns the errors. For the same reason, we require the OrgEnrollmentConfigRequest object. We will end up writing the same code with our own constraint or through propertyeditors or converters (which are anyways unusable in this rest api context).